src/StoreApi/Utilities/CartController.php

```diff
@@ -360,15 +360,78 @@ private function get_error_message_for_stock_exception_type( $exception_type, $s
 		return __( 'There was an error with an item in your cart.', 'woo-gutenberg-products-block' );
 	}
 
+	/**
+	 * Validate cart and check for errors.
+	 *
+	 * @throws InvalidCartException Exception if invalid data is detected in the cart.
+	 * @throws RouteException Exception if invalid data is detected in the cart.
+	 */
+	public function validate_cart() {
+
+		$cart = $this->get_cart_instance();
+		$this->validate_cart_items();
+		$this->validate_cart_coupons();
+
+		/**
+		 * Fire action to validate cart. Functions hooking into this should return the WP_Error enchanced with custom errors.
+		 *
+		 * @example See docs/examples/validate-cart.md
+		 *
+		 * @param WP_Error $errors  WP_Error object.
+		 * @param WC_Cart  $cart    Cart object.
+		 *
+		 * @return WP_Error         The WP_Error object returned.
+		 */
+		$cart_errors = apply_filters( '__experimental_woocommerce_store_api_cart_errors', new WP_Error(), $cart );
+
+		if ( ! is_wp_error( $cart_errors ) ) {
+
+			throw new RouteException( 'missing_cart_validation_server_error', __( 'There was an error with your cart.', 'woo-gutenberg-products-block' ) );
+		}
+
+		if ( $cart_errors->has_errors() ) {
```

Comment:
Based on the above comment, this would need to be wrapped part of the `is_wp_error` conditional.